stl/inc/future

```diff
@@ -402,7 +425,9 @@ private:
 
     virtual void _Do_notify(unique_lock<mutex>* _Lock, bool _At_thread_exit) { // notify waiting threads
         // TRANSITION, ABI: This is virtual, but never overridden.
-        _Has_stored_result = true;
+        if (is_default_constructible<_Ty>::value) {
+            _Has_stored_result = true;
+        }

```

Comment:
May be better to use `if constexpr`.
```suggestion
        if constexpr (is_default_constructible_v<_Ty>) {
            _Has_stored_result = true;
        }
```

Reply:
Yep, this would be a slight improvement for code clarity and debug codegen. (In non-optimized debug mode, a branch will be emitted at runtime for `if`, but not for `if constexpr`.)